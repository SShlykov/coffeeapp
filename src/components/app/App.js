import React, { Component } from "react";
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import FirstScreen from "../FirstScreen";
import {Contacts, ForYourPleasure, MainPage, OurCoffee, ProductPage} from "../Pages"
import Footer from  "../Footer";

export default class App extends Component {
  render() {

    return (
      <Router>
          <>
              <Switch>
                  <Route exact path="/" component={FirstScreen}/>
                  <Route path="/ourcoffee/" component={() => {return <FirstScreen heading="Our coffee"/>}}/>
                  <Route path="/foryourpleasure/" component={() => {return <FirstScreen heading="For your pleasure"/>}}/>
                  <Route path="/contacts/" component={() => {return <FirstScreen heading="Contact Us"/>}}/>
              </Switch>
              <Switch>
                  <Route exact path="/" component={MainPage}/>
                  <Route exact path="/ourcoffee/" component={OurCoffee}/>
                  <Route exact path="/ourcoffee/:id" component={({match}) => {
                      const {id} = match.params;
                      return <ProductPage id={id} />
                  }}/>
                  <Route exact path="/foryourpleasure/" component={ForYourPleasure}/>
                  <Route exact path="/contacts/" component={Contacts}/>
              </Switch>
              <Footer/>
          </>
      </Router>
    );
  }
}
