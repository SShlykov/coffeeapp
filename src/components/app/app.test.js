import React from "react";
import App from "./app";
import  {shallow} from "enzyme";

describe("Testing <App />", () => {
    const coffeeApp = shallow(<App />);
    describe("Testing snap and state", () => {
        it("App gave rendered correctly", () => {
            expect(coffeeApp).toMatchSnapshot();
        });
        it("App state 'error' is true", () => {
            expect(coffeeApp.state().error).toBeFalsy();
        });
        it("App state 'term' is empty string", () => {
            expect(coffeeApp.state().filter).toBe("");
        });
        it("App state 'filter' is empty string", () => {
            expect(coffeeApp.state().filter).toBe("");
        });

    })
    describe("Testing handlers", () => {
        it("Testing onSearchChange", () => {
            coffeeApp.instance().onSearchChange("hello");
            expect(coffeeApp.state().term).toBe("hello");
        })
        it("Testing onFilterChange", () => {
            coffeeApp.instance().onFilterChange("world");
            expect(coffeeApp.state().filter).toBe("world");
        })
    })
});