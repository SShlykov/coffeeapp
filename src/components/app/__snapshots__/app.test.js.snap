// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <App /> Testing snap and state App gave rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <React.Fragment>
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/ourcoffee/"
          />
          <Route
            component={[Function]}
            path="/foryourpleasure/"
          />
          <Route
            component={[Function]}
            path="/contacts/"
          />
        </Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          exact={true}
          path="/ourcoffee/"
          render={[Function]}
        />
        <Route
          component={[Function]}
          exact={true}
          path="/ourcoffee/:id"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/foryourpleasure/"
        />
        <Route
          exact={true}
          path="/contacts/"
          render={[Function]}
        />
        <Footer />
      </React.Fragment>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/ourcoffee/"
            />
            <Route
              component={[Function]}
              path="/foryourpleasure/"
            />
            <Route
              component={[Function]}
              path="/contacts/"
            />
          </Switch>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            exact={true}
            path="/ourcoffee/"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/ourcoffee/:id"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/foryourpleasure/"
          />,
          <Route
            exact={true}
            path="/contacts/"
            render={[Function]}
          />,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                path="/ourcoffee/"
              />,
              <Route
                component={[Function]}
                path="/foryourpleasure/"
              />,
              <Route
                component={[Function]}
                path="/contacts/"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/ourcoffee/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/foryourpleasure/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/contacts/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/ourcoffee/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/ourcoffee/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/foryourpleasure/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/contacts/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <React.Fragment>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/ourcoffee/"
            />
            <Route
              component={[Function]}
              path="/foryourpleasure/"
            />
            <Route
              component={[Function]}
              path="/contacts/"
            />
          </Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            exact={true}
            path="/ourcoffee/"
            render={[Function]}
          />
          <Route
            component={[Function]}
            exact={true}
            path="/ourcoffee/:id"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/foryourpleasure/"
          />
          <Route
            exact={true}
            path="/contacts/"
            render={[Function]}
          />
          <Footer />
        </React.Fragment>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                path="/ourcoffee/"
              />
              <Route
                component={[Function]}
                path="/foryourpleasure/"
              />
              <Route
                component={[Function]}
                path="/contacts/"
              />
            </Switch>,
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              exact={true}
              path="/ourcoffee/"
              render={[Function]}
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/ourcoffee/:id"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/foryourpleasure/"
            />,
            <Route
              exact={true}
              path="/contacts/"
              render={[Function]}
            />,
            <Footer />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <Route
                  component={[Function]}
                  path="/ourcoffee/"
                />,
                <Route
                  component={[Function]}
                  path="/foryourpleasure/"
                />,
                <Route
                  component={[Function]}
                  path="/contacts/"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/ourcoffee/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/foryourpleasure/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/contacts/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/ourcoffee/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/ourcoffee/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/foryourpleasure/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/contacts/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
